src/database/DatabaseConnection.ts

```diff
@@ -1,7 +1,7 @@
 
 export interface DatabaseConnection {
   getConnection(): Promise<DatabaseConnection>;
-  query<T = void>(sql: any, parameters?: any[]): any;
+  query<RowType = any>(sql: any, parameters?: any[]): Promise<[RowType[], any]>;
```

Comment:
I wonder if unknown would be more appropriate

Reply:
There are a number of queries which haven't had a type argument added. I am happy to fill them in if you want.